### ISSUE TITLE
eth_sendTransaction does not return anything at all

### ISSUE THREAD

**Issue by jesuscript** [ethereum/cpp-ethereum#1959]
With a contract function like:
```
  function issue(address recipient, uint32 amount){
    if(msg.sender == owner){
      balance[recipient] += amount;
    }
  }
```

When making a call from nodejs using `c.issue.sendTransaction(settings.addresses[0], 1,{from: web3.eth.coinbase}, function(err, result){`, the `result` is an empty string.

A synchronous call produces the same result.

Here are the request payload and the response for the sendTransaction RPC call:

```
REQ: { jsonrpc: '2.0',
  method: 'eth_sendTransaction',
  params:
   [ { from: '0x45cc4ec50fb12da5d4e24bb5319de6b9b8488097',
       to: '0x779becdbcc413489409a5f4081dd4f4b5a718d22',
       data: '0x3284fd790000000000000000000000003c9958527426ae274ea973bb82247972d19b43cd0000000000000000000000000000000000000000000000000000000000000001' } ],
  id: 5 }

RES:  { id: 5, jsonrpc: '2.0', result: '' }
```

I am not 100% sure what the actual expected result should be, but an empty string seems wrong. Also, this section of JS API docs: https://github.com/ethereum/wiki/wiki/JavaScript-API#contract-methods says

**Returns**

* **String** - If its a call the result data, if its a send transaction a created contract address, or the transaction hash, see web3.eth.sendTransaction for details.*

According to this, the result should at least contain the transaction hash

**Comment by ethers** [comment]
This fix would be really good to have because without it, looking into what transaction has been sent is very difficult: leading to other issues/suggestions #1837, #1999, #2000

**Comment by debris** [comment]
fixed in #2404
